Modules/_functoolsmodule.c

```diff
@@ -626,10 +626,11 @@ functools_reduce(PyObject *self, PyObject *args)
         if (result == NULL)
             result = op2;
         else {
-            PyTuple_SetItem(args, 0, result);
-            PyTuple_SetItem(args, 1, op2);
-            if ((result = PyEval_CallObject(func, args)) == NULL)
+            Py_XSETREF(PyTuple_GET_ITEM(args, 0), result);
+            Py_XSETREF(PyTuple_GET_ITEM(args, 1), op2);
```

Comment:
Does `PyTuple_GET_ITEM` guarantee its result is an lvalue?

Reply:
I fixed it, `PySequence_Fast_ITEMS()` seems to be slower as it checks for type.